File: HRMS.Core/Models/Searches/EventSearch.cs
namespace HRMS.Core.Models.Searches;

public class EventSearch : BaseSearch
{
    public string? Name { get; set; }

    public int? EmployeeId { get; set; }

    public bool IncludeEventType { get; set; }

    public bool IncludeEmployee { get; set; }
}

File: HRMS.Desktop/lib/models/searches/event_search.dart
import 'package:json_annotation/json_annotation.dart';

import 'base_search.dart';

part 'event_search.g.dart';

@JsonSerializable()
class EventSearch extends BaseSearch {
  String? name;
  int? employeeId;
  bool includeEventType = false;
  bool includeEmployee = false;  

  EventSearch();

  factory EventSearch.fromJson(Map<String, dynamic> json) =>
      _$EventSearchFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$EventSearchToJson(this);
}
